src/base/http/requestparser.cpp

```diff
@@ -208,7 +208,7 @@ bool RequestParser::parseRequestLine(const QString &line)
     const int sepPos = url.indexOf('?');
     const QByteArrayView pathComponent = ((sepPos == -1) ? url : QByteArrayView(url).mid(0, sepPos));
 
-    m_request.path = QString::fromUtf8(QByteArray::fromPercentEncoding(pathComponent.toByteArray()));
+    m_request.path = QString::fromUtf8(QByteArray::fromPercentEncoding(fromView(pathComponent)));
```

Comment:
>I would even use QByteArray::fromRawData(view.constData(), view.size()) directly since QByteArray::fromRawData() is well-known (explicitly declared) as producing "weak" QByteArray that doesn't own the data.

A slight advantage of having helper function is that it only needs one parameter and in some cases, the code will become shorter than using `QByteArray::fromRawData`.

Reply:
Not a big problem anyway.
And Yes, helper is still useful in case of temporary objects, like `asQByteArray(list[0])`.